src/functions/edhMarshaller.ts

```diff
@@ -45,14 +45,19 @@ const edhMarshaller: Handler = async (event: GetRecordsOutput, context?: Context
     });
 
     return Promise.all(sendMessagePromises)
-    .catch((error: AWSError) => {
-        console.error(error);
-        console.log("records");
-        console.log(records);
-        if(error.code !== "InvalidParameterValue"){
-            throw error;
-        }
-    });
+        .catch((error: AWSError) => {
+            console.error(error);
+            console.log("records");
+            console.log(records);
+            // do not retry 4XX error codes, except 429. Retry only 500 and 429.
+            if ((error.statusCode >= 400 && error.statusCode < 500) && error.statusCode !== 429) {
+                // return something so the Lambda won't retry
+                return;
```

Comment:
The ticket specifies the following in the Acceptance Criteria: "Other 4XX errors should be logged as errors both the specific error and the PII redacted payload that caused it, and not retried. These messages should go directly to the DLQ."
If you just return here, we will not be able to keep track of the messages that weren't sent successfully

Reply:
There are 3 console.logs that will log the errors before this IF condition and we can see what kind of error it is

Reply:
The acceptance criteria also mentions "These messages should go directly to the DLQ" and I can't see the logic for it

Reply:
The idea is that if the lambda throws any errors, it will automatically retry. However, if you return something, the execution is considered successful and it won't retry. We cannot manually send messages to DLQ, it is automatically handled by AWS when a message is retried multiple times and hits maxReceiveCount, but from what I understood we want to avoid retries if the lambda throws an error. 

So we want to avoid retries, but a message will be sent to DLQ only if it's retried multiple times...looks like these 2 requests are conflicting.